peer-monitoring-service/client/src/tests/single_peer.rs

```diff
@@ -415,6 +434,72 @@ async fn test_latency_ping_failures() {
     verify_peer_latency_state(&peer_monitor_state, &validator_peer, 3, 0);
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn test_disconnect_from_peer() {
+    // Create the peer monitoring client and server
+    let network_id = NetworkId::Validator;
+    let (peer_monitoring_client, mut mock_monitoring_server, peer_monitor_state, time_service) =
+        MockMonitoringServer::new(vec![network_id]);
+
+    let peers = peer_monitoring_client.get_peers_and_metadata();
+
+    // Create a node config where only latency pings refresh
+    // Note: By default, this will have a max latency ping tolerated of 3
```

Comment:
nit: I would set this directly and then reference the variable below (e.g., in case we change the default value):
```
// Create a node config where only latency pings refresh
let mut node_config = config_with_latency_ping_requests();

// Set the maximum number of ping failures
let max_latency_ping_failures = 3;
node_config.peer_monitoring_service.latency_monitoring.max_latency_ping_failures = max_latency_ping_failures;

...

for i in 0..max_latency_ping_failures { ... }
``